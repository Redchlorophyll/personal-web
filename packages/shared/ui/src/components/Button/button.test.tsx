import { Button } from './index';
import { useState } from 'react';
import {
  renderWithProviders,
  screen,
  userEvent,
  act,
} from 'testing-library-react-custom';

function ButtonWrapper() {
  const [status, setStatus] = useState<string>('empty');

  function handleOnClick() {
    setStatus('clicked');
  }

  return (
    <div>
      <p>{status}</p>
      <Button type={'solid'} variant={'warning'} onClick={handleOnClick}>
        Test
      </Button>
    </div>
  );
}

describe('shared - ui - Button', () => {
  test('should render properly', () => {
    const { baseElement } = renderWithProviders(<ButtonWrapper />);

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  test('It should render properly', () => {
    renderWithProviders(<ButtonWrapper />);
    const target = screen.getByText('Test');

    expect(target).toBeInTheDocument();
  });

  test('It should show status when clicked', async () => {
    const user = userEvent.setup();
    renderWithProviders(<ButtonWrapper />);
    const target = screen.getByRole('button');

    expect(screen.queryByText('empty')).toBeInTheDocument();

    await act(async () => await user.click(target));

    expect(screen.queryByText('clicked')).toBeInTheDocument();
  });
});
