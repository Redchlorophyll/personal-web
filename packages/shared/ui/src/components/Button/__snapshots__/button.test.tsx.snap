// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared - ui - Button should render properly 1`] = `
<body>
  <div>
    <div>
      <p>
        empty
      </p>
      <button
        class="w-fit min-w-[140px] px-[27px] pt-[3px] pb-[6px] h-[34px] rounded-[27.2748px] leading-[22px] bg-yellow-700 text-black-100 active:bg-yellow-900"
        type="button"
      >
        Test
      </button>
    </div>
  </div>
</body>
`;
